File: _FFinal/p10400_The3n+1Problem.java
//要用long，這題用 c++ 會比較好，吃的時間和記憶體比較少

import java.util.*;

class FFtest{

	static long add(long a){
		long len = 1;
		while(a != 1){
			if(a % 2 != 0) a = 3 * a + 1;
			else a /= 2;
			len++;
		}
		return len;
	}

	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);

		while(sc.hasNext()){
			long a = sc.nextLong();
			long b = sc.nextLong();

			System.out.printf("%d %d ", a, b);

			if(a > b){
				long tmp = a;
				a = b;
				b = tmp;
			}

			long maxLen = 0;
			for(long i = a; i<=b; i++){
				maxLen = Math.max(maxLen, add(i));
			}

			System.out.println(maxLen);
		}

	}
}

File: _FFinal/p10400_The3n+1Problem.cpp
//要用 long 才可以，如果用 int 會 TLE
#include <iostream>
#include <cstring>

using namespace std;

long add(long a){
	long len = 1;
	while(a!=1){
		if(a%2!=0) a = 3 * a + 1;
		else a /= 2;
		len++;
	}

	return len;
}

int main(){
	long a, b;
	while(cin >> a >> b){

		cout << a << " " << b << " ";
		
		if(a > b){
			long tmp = a;
			a = b;
			b = tmp;
		}

		long maxLen = 0;
		for(long i=a; i<=b; i++){
			maxLen = max(maxLen, add(i));
		}
		cout << maxLen << endl;
	}
}